src/NationalInstruments.Analyzers/Correctness/DisposableOmitsFinalizerAnalyzer.cs

```diff
@@ -0,0 +1,51 @@
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using NationalInstruments.Analyzers.Properties;
+using NationalInstruments.Analyzers.Utilities;
+using NationalInstruments.Analyzers.Utilities.Extensions;
+
+namespace NationalInstruments.Analyzers.Correctness
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class DisposableOmitsFinalizerAnalyzer : NIDiagnosticAnalyzer
+    {
+        public const string DiagnosticId = "NI1816X";
+
+        public static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            DiagnosticId,
+            new LocalizableResourceString(nameof(Resources.NI1816X_Title), Resources.ResourceManager, typeof(Resources)),
+            new LocalizableResourceString(nameof(Resources.NI1816X_MessageFormat), Resources.ResourceManager, typeof(Resources)),
+            category: "NationalInstruments",
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: true,
```

Comment:
Probably should make this `false` for the time being

Reply:
Based on the other conversation I'm marking it disabled by default. Someone can show me how to enable it when we pull it into ASW.